### ISSUE TITLE
ESLint Plugin: Flat Config Support

### ISSUE THREAD

**Issue by davidjbng** [TanStack/query#7544]
### Describe the bug

eslint-plugin-query does not yet support the new Flat Config format

### Your minimal, reproducible example

https://github.com/davidjbng/eslint-plugin-query-v9-example

### Steps to reproduce

1. Checkout https://github.com/davidjbng/eslint-plugin-query-v9-example
2. Follow https://eslint.org/docs/latest/use/configure/migration-guide to add Flat Config support

### Expected behavior

eslint-plugin-query works with Flat Config

### How often does this bug happen?

None

### Platform

not relevant

### Tanstack Query adapter

None

### TanStack Query version

v5.40.1

### Additional context

ESLint V9 support is adressed in https://github.com/TanStack/query/pull/7543

I did some inital work to export a `flat/recommended` config earlier which seemed to work, but a bit more testing was still needed. See https://github.com/TanStack/query/compare/main...davidjbng:query:eslint-v9-flat-config

## Resources
- https://eslint.org/docs/latest/use/configure/migration-guide
- https://eslint.org/docs/latest/extend/plugin-migration-flat-config#migrating-rules-for-flat-config

**Comment by SimenB** [comment]
Workaround, if people need it:

```js
import { fixupPluginRules } from '@eslint/compat';
import * as eslintPluginTanstackQuery from '@tanstack/eslint-plugin-query';

export default [
  {
    plugins: {
      '@tanstack/query': fixupPluginRules(eslintPluginTanstackQuery),
    },
    rules: eslintPluginTanstackQuery.configs.recommended.rules,
  },
];

```

**Comment by lachlancollins** [comment]
Fixed in #7663, see https://tanstack.com/query/latest/docs/eslint/eslint-plugin-query for setup instructions
